### PR TITLE
feat: Add VS 2022 .NET 6 template, split wizard

### LINKED ISSUE #7737: [Wasm] Debugger is not configured by default
### Current behavior

- create solution template as per [here](https://platform.uno/docs/articles/get-started-winui3.html)
- start VS 2022 and update to latest NuGet packages
- place breakpoint somewhere in App.xaml.cs
- build and start WASM

-> breakpoint not hit and tons of errors in browser console

This makes it impossible to use the template as a reference to e.g. iron out issues when migrating from older UNO versions :-(

> ​ fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint [fe80::1dbb:75df:6d18:3fe3].
eval @ unknown
​ fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint 192.168.178.64.
eval @ unknown
​ fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint [fe80::2d92:3431:e709:9cfd].
eval @ unknown
​ fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint 169.254.156.253.
eval @ unknown
​ fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint [fe80::51a8:9d4:f975:115d].
eval @ unknown
​ fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint 169.254.17.93.
eval @ unknown
​ fail: Uno.UI.RemoteControl.RemoteControlClient[0]
      Failed to get remote control server port from the IDE for endpoint [fe80::d4fb:71cd:dc60:a1c0].

### Expected behavior

Building a WASM project from suggested template should work no problem

### How to reproduce it (as minimally and precisely as possible)

Above

### Workaround

Unknown

### Works on UWP/WinUI

No

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

Latest

### Affected platforms

WebAssembly

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): closes #7737, closes https://github.com/unoplatform/uno/issues/5447, closes https://github.com/unoplatform/uno/issues/4258, closes https://github.com/unoplatform/uno/issues/3885, closes https://github.com/unoplatform/uno/issues/6669, closes https://github.com/unoplatform/uno/issues/806, closes https://github.com/unoplatform/uno/issues/5009, closes https://github.com/unoplatform/uno/issues/5486

# Significant changes of this PR are in the wizard, here: https://github.com/unoplatform/uno/pull/8762/files#diff-b69961baf38fcfd3f1eb572d4fc6c2ffdb69a1724a92c61d9fb4f49c1f7c505aR19, the rest is vsix template changes.

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

- Wasm Debugger support is now enabled by default
- Default template uses .NET Mobile + WinUI + WinAppSDK
- Platform selection UI to create customized templates
- Don't allow the creation of a project with unsupported names (e.g. leading numbers)
- New WinUI project library

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/SamplesApp.Wasm/Properties/launchSettings.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
     "windowsAuthentication": false,
     "anonymousAuthentication": true,
     "iisExpress": {
-      "applicationUrl": "http://localhost:52820/",
-      "sslPort": 44375
+      "applicationUrl": "http://localhost:51085/",
+      "sslPort": 44306
```
Comment (MartinZikmund) [inline src/SamplesApp/SamplesApp.Wasm/Properties/launchSettings.json:7]: Are these intentional?

File: src/SolutionTemplate/UnoLibraryTemplate.net6/CrossTargetedLibrary.vstemplate [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
-	<ProjectTypeTag>Uno Platform</ProjectTypeTag>
-  </TemplateData>
-  <TemplateContent>
-	<Project TargetFileName="$ext_safeprojectname$.csproj" File="CrossTargetedLibrary.csproj" ReplaceParameters="true">
-	  <ProjectItem ReplaceParameters="true" OpenInEditor="true">Class1.cs</ProjectItem>
-	</Project>
-  </TemplateContent>
+	<TemplateData>
+		<Name>Uno Platform Library</Name>
+		<Description>Template for creating a multi-platform library with the Uno Platform that targets WinUI, WebAssembly, Android, iOS, macOS, Mac Catalyst and Linux</Description>
```
Comment (MartinZikmund) [inline src/SolutionTemplate/UnoLibraryTemplate.net6/CrossTargetedLibrary.vstemplate:4]: Does it mean there won't be a .NET 6 template with UWP 🤔 ?

**Review by MartinZikmund — APPROVED** [review]
Let's gooooo! 🚀

File: src/SamplesApp/SamplesApp.Wasm/Properties/launchSettings.json (hunk 1 above)
Comment (jeromelaban) [reply src/SamplesApp/SamplesApp.Wasm/Properties/launchSettings.json:7]: VS changes those automatically when there are conflicts now... I'll remove

**Pull request merged**